src/applications/microservices/payforadoption-go/database.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+package main
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/url"
+
+	"petadoptions/payforadoption"
+
+	"github.com/XSAM/otelsql"
+	"github.com/go-kit/log"
+	_ "github.com/lib/pq"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.26.0"
+)
+
+// createInstrumentedDB creates an instrumented database connection
+// Aurora correlation is handled by the SQL span processor
+func createInstrumentedDB(ctx context.Context, cfg payforadoption.Config) (*sql.DB, error) {
+	// Get database configuration from secrets manager
+	dbService := payforadoption.NewDatabaseConfigService(cfg)
+	dbConfig, err := dbService.GetDatabaseConfig(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get database config: %w", err)
+	}
+
+	// Build connection string
+	connStr := buildConnectionString(dbConfig)
+
+	// Build resource identifier for Aurora correlation
+	// Format: engine|host|port for CloudWatch Application Signals correlation
+	resourceIdentifier := fmt.Sprintf("postgres|%s|%d", dbConfig.Host, dbConfig.Port)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential log injection detected. Ensure all untrusted input is properly sanitized before logging. Use parameterized logging or validate input against an allow list to prevent log injection vulnerabilities. Consider using a dedicated logging library&#39;s built-in sanitization features when available. Learn more - https://cwe.mitre.org/data/definitions/117.html